PlutoHelperAgent/AppDelegate.m

```diff
@@ -125,92 +125,150 @@ -    (void)getUrl:(NSAppleEventDescriptor *)event
-        [ProjectLockerAndKeychainFunctions get_data_from_server:@"/api/project/"
-                                                               :@"/files"
-                                                               :projectid
-                                              completionHandler:^void (NSURLResponse *response,NSDictionary *filesResult){
-            [ProjectLockerAndKeychainFunctions get_data_from_server:@"/api/storage/"
-                                                                   :NULL
-                                                                   :filesResult[@"files"][0][@"storage"]
-                                                  completionHandler:^(NSURLResponse *response, NSDictionary *storageResult) {
-                if (storageResult[@"result"][@"clientpath"] == NULL) {
-                    NSAlert *alert = [[NSAlert alloc] init];
-                    
-                    [alert addButtonWithTitle:@"Okay"];
-                    
-                    NSString *message = [NSString stringWithFormat: @"No client path on storage ID %@", filesResult[@"files"][0][@"storage"]];
-                    
-                    [alert setMessageText:message];
-                    
-                    [alert setInformativeText:@"Can't open project, because it's on a storage which has no client path set.\n\nPlease contact multimediatech@theguardian.com."];
-                    
-                    [alert setAlertStyle:NSWarningAlertStyle];
-                    
-                    if ([alert runModal] == NSAlertFirstButtonReturn) {
-                        
-                    }
-                    
-                } else {
-                    NSAlert *alert = [[NSAlert alloc] init];
-                    NSString *helperScript = [[NSUserDefaults standardUserDefaults] stringForKey:@"local_shell_script"];
-                    
-                    if([helperScript compare:@""]==0){
-                        dispatch_async(dispatch_get_global_queue(0, 0), ^{
-                            [alert setMessageText:@"Setup problem"];
-                            [alert setInformativeText:@"Your mac does not appear to be set up correctly, no helper script is configured. Please contact multimediatech@theguardian.com."];
+        [ProjectLockerAndKeychainFunctions check_logged_in:^(enum ReturnValues connectionStatus) {
+            
+            if (connectionStatus == ALLOK) {
+                [ProjectLockerAndKeychainFunctions get_data_from_server:@"/api/project/"
+                                                                       :@"/files"
+                                                                       :projectid
+                                                      completionHandler:^void (NSURLResponse *response,NSDictionary *filesResult){
+                    [ProjectLockerAndKeychainFunctions get_data_from_server:@"/api/storage/"
+                                                                           :NULL
```

Comment:
there are  a lot of indented blocks here which makes the diff really hard to read.  Please could you seperate into different methods

e.g.

```
switch(connectionStatus){
  case ALLOK:
    [self tryOpenProject];
    break;
  case PERMISSION_DENIED:
    [self showError:@"Permission Denied" informativeText:@"An unknown error occurred. Please report this error to multimediatech@theguardian.com" showPrefs:YES];
    break;
  case SERVER_ERROR:
    [self showError:@"Server ERror" informativeText:@"An error occurred on the server. Please report this error to multimediatech@theguardian.com" showPrefs:NO];
    break;
  default:
   [self showError@"Unknown error" ......
}
```

then put the stuff that responds to ALLOK into a method called tryOpenProject, the code to show error messages into a method called showError, etc.